### PR TITLE
update license/readme URLs to HTTPS where appropriate

### PR DESCRIPTION
**Pull request by rrennick**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This updates woocommerce.com, fsf.org, and gnu.org URLs in the license & readme files to be HTTPS. As this isn't changing any code I didn't create an issue for it.

### CODE REVIEW

**Review by rodrigoprimo — COMMENTED** [review]
Thanks for paying attention to this detail. I think that there is just one minor problem that we need to fix on the ideas.woocommerce.com side before we can merge this PR.

https://ideas.woocommerce.com uses a certificate that is not valid for its domain. I will check to see who can help us with that. If it is not easy to fix this issue (we use a third-party service for the ideas board), I suggest we remove it from the list of changes in this PR.

File: readme.txt [hunk 1]
```diff
@@ -126,7 +126,7 @@ Yes! WooCommerce will work with any theme, but may require some styling. Please
 
 = Where can I request new features, eCommerce themes, and extensions? =
 
-You can vote on and request new features and extensions on our [WooIdeas board](http://ideas.woocommerce.com/forums/133476-woocommerce?utm_source=wp%20org%20repo%20listing&utm_content=3.6)
+You can vote on and request new features and extensions on our [WooIdeas board](https://ideas.woocommerce.com/forums/133476-woocommerce?utm_source=wp%20org%20repo%20listing&utm_content=3.6)
```
Comment (kloon) [inline readme.txt]: The ideas board link needs to stay http as our plan with the service provider does not include https.

**Review by kloon — CHANGES_REQUESTED** [review]
Just the ideas link need to be kept the same then this is good to go.

File: readme.txt (hunk 1 above)
Comment (rrennick) [reply readme.txt]: @rodrigoprimo @kloon Thanks, I should have checked that subdomain first. I've reverted that URL.

**Review by kloon — APPROVED** [review]
Thanks, Ron!

**Pull request merged**